routes/buyer/buyer.js

```diff
@@ -136,6 +136,62 @@ exports.updateBuyerById = function(req, res) {
     }
 };
 
+
+/**
+
+Update Buyer Demand
+
+
+ **/
+
+
+exports.editDemand = function(req, res) {
+    if(common.isAuthenticated(req, res)){
+
+        Demand.findByIdAndUpdate({_id:req.params.demand_id}, req.body, function (err, changes) {
+            console.log(err, changes);
+            if (err) {
+                handleDBError(err, res);
+            } else {
+                res.send(changes);
+            }
+        });
+
+    }
+};
+
+exports.editRep = function(req, res) {
+    if(common.isAuthenticated(req, res)){
+
+        Buyer.findById(req.params.id, function (err, doc) {
+
+            var this_rep = doc.re_representatives.id(req.params.rep_id);
+
+            this_rep.re_email = req.body.re_email;
+
+            this_rep.re_contact = req.body.re_contact;
+
+            this_rep.re_name = req.body.re_name;
+
+            this_rep.re_crop = req.body.re_crop;
+
+            this_rep.re_security_question = req.body.re_security_question;
+
+            this_rep.re_security_answer = req.body.re_security_answer;
+
+            doc.save();
+
+            res.status(200);
```

Comment:
give `doc.save();` a callback function reports whether the save function was successful or not and report it to the user.